packages/swap/src/event-handlers/depositIgnoredV120.ts

```diff
@@ -0,0 +1,90 @@
+import { type Chain } from '.prisma/client';
+import { encodeAddress } from '@polkadot/util-crypto';
+import { z } from 'zod';
+import { encodeAddress as encodeBitcoinAddress } from '@/shared/bitcoin';
+import { Asset, assetChains } from '@/shared/enums';
+import {
+  DOT_PREFIX,
+  chainflipAssetEnum,
+  hexString,
+  rustEnum,
+  u128,
+} from '@/shared/parsers';
+import env from '../config/env';
+import type { EventHandlerArgs } from './index';
+
+const reasonSchema = rustEnum(['BelowMinimumDeposit', 'NotEnoughToPayFees']);
+
+type Reason = z.output<typeof reasonSchema>;
+
+const depositIgnoredArgs = z
+  .object({
+    asset: chainflipAssetEnum,
+    amount: u128,
+    depositAddress: z.union([
+      z
+        .object({ __kind: z.literal('Taproot'), value: hexString })
+        .transform((o) => {
+          try {
+            return encodeBitcoinAddress(o.value, env.CHAINFLIP_NETWORK);
+          } catch {
+            return null;
+          }
+        }),
+      hexString,
+    ]),
+    reason: reasonSchema,
```

Comment:
v120 adds `reason`. I initially added this to the original handler with a default reason, but then I realized that if this field is removed, we wouldn't catch that error and would silently use the default value which may be wrong.